src/hotspot/share/classfile/klassFactory.cpp

```diff
@@ -132,8 +132,8 @@ static ClassFileStream* check_class_file_load_hook(ClassFileStream* stream,
 
     if (state != NULL) {
       Klass* k = state->get_class_being_redefined();
-
-      if (k != NULL) {
+      // use cached_class_file only for RetransformClasses
```

Comment:
More detailed comment would be nice to have here to explain in what scenario this change is needed.

Reply:
I updated a comment before the code block and removed this one